### PR TITLE
Improve Matching

### PR DESCRIPTION
**Pull request by rodneykinney**
Matching and parsing fixes

### CODE REVIEW

File: src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala [hunk 1]
```diff
@@ -14,17 +16,25 @@ object BibliographyErrorAnalysis {
...
-      Title(data.title.normalized.text.splitOnWhitespace.head),
-      Venue(""),
+    FuzzyPaperMetadata(
+      Title(data.title.normalized.text),
       data.year,
-      data.authors.map(_.lastNameOnly.normalized).take(2)
+      data.authors.map(_.lastNameOnly.normalized).toSet
     )
 
+  case class FuzzyPaperMetadata(title: Title, year: Year, authors: Set[Author]) {
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala:30]: Extracted bibliography entries are considered a match if either the normalized title matches or the authors and year matches.

File: src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala [hunk 2]
```diff
@@ -14,17 +16,25 @@ object BibliographyErrorAnalysis {
...
+    override def equals(a: Any) = a match {
+      case FuzzyPaperMetadata(title2, year2, authors2) =>
+        title == title2 ||
+          (authors == authors2 && year == year2)
+      case _ => false
+    }
+    override def hashCode = 1
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala:37]: Hashcode must be a constant to be consistent with equals :(

File: src/main/scala/org/allenai/scholar/metrics/metadata/Parser.scala [hunk 3]
```diff
@@ -71,7 +71,7 @@ abstract class Parser(
     * @return The paper's core metadata.
     */
   def parseCoreMetadataString(xmlString: String): MetadataAndBibliography = {
-    val doc = Jsoup.parse(xmlString)
+    val doc = Jsoup.parse(xmlString, "", org.jsoup.parser.Parser.xmlParser())
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/scholar/metrics/metadata/Parser.scala:74]: This seems to fix the problem with JSoup parsing `<title/>` incorrectly.  Yay!

File: src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala (hunk 1 above)
Comment (rodneykinney) [inline src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala:30]: We should start using this logic to build the citation graph.

File: src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala [hunk 4]
```diff
@@ -14,17 +16,25 @@ object BibliographyErrorAnalysis {
 
   val BIBLIOGRAPHY_EXTRACTION = "bibliography"
 
-  def extractBibliography(bibs: Set[PaperMetadata]): Iterable[PaperMetadata] =
+  def extractBibliography(bibs: Set[PaperMetadata]): Iterable[FuzzyPaperMetadata] =
```
Comment (vha14) [inline src/main/scala/org/allenai/scholar/metrics/metadata/BibliographyErrorAnalysis.scala:19]: Can `bibs` just be an `Iterable` instead of `Set`?

**Comment by jakemannix** [comment]
ok, shipit!

**Pull request merged**